### ISSUE TITLE
Local/custom refactorings

### ISSUE THREAD

**Issue by markusjohnsson** [krizzdewizz/vscode-refactorix#3]
Great extension, thanks for sharing!

As I understand it, if I want to implement a new refactoring right now I would have to re-package the extension, right? Are there any plans to allow local or custom refactorings?

**Comment by krizzdewizz** [comment]
Hi and thanks!
No, I do not plan this as a framework for custom refactorings. But instead of re-packaging, an option would be to include it in refactorix directly. I'm open for new stuff and contributions are welcome! What feature(s) do you have in mind?

**Comment by markusjohnsson** [comment]
Yeah, integrating into the main package makes sense most of the time. I had a couple of refactorings in mind:

- Extract/move to file - move the current selection, or the declaration directly under the cursor, to its own file and import it to the current file
- Reverse `if` - invert boolean expression and swap then/else blocks
- Extract method/function - move selected statements into new method/function
